### PR TITLE
SubscriptionBuilder

### PR DESCRIPTION
**Pull request by jhorstmann**
The overloads for the `subscribe` method were getting a bit out of hand.

Also includes some small fixes to log the correct event name when subscribing to multiple events.

### CODE REVIEW

**Comment by jhorstmann** [comment]
@AdnanKukuljac interested in your review

**Comment by AdnanKukuljac** [comment]
@jhorstmann I like the idea of SubscriptionBuilder. It's much cleaner.

**Comment by AdnanKukuljac** [comment]
👍

**Comment by jhorstmann** [comment]
:+1:

File: fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java [hunk 1]
```diff
@@ -85,62 +86,32 @@ public static NakadiClientBuilder builder(URI baseUri) {
...
-     * @param eventName
-     * @param consumerGroup
-     * @param readFrom
-     * @return
-     * @throws IOException
+     * Build a subscription for a single event type.
      */
-    public Subscription subscribe(String applicationName, String eventName, String consumerGroup, SubscriptionRequest.Position readFrom, List<Cursor> initialCursors) throws IOException {
-        return subscribe(applicationName,  Collections.singleton(eventName), consumerGroup, readFrom, initialCursors);
+    public SubscriptionBuilder subscribe(String applicationName, String eventName) throws IOException {
```
Comment (lukasniemeier-zalando) [inline fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java]: Or name those `subscription` and the terminal operation `subscribe`, i.e. clients will have `.subscription(applicationName, eventName).withConsumerGroup(group).subscribe()`

**Comment by jhorstmann** [comment]
:+1:

**Pull request merged**